src/daemon/control.rs

```diff
@@ -505,7 +513,7 @@ pub fn check_spend_signatures(
             let sig = psbtin
                 .partial_sigs
                 .get(&pubkey)
-                .ok_or_else(|| SigError::MissingSignature(pubkey))?;
+                .ok_or(SigError::MissingSignature(pubkey))?;
```

Comment:
hey, that's clippy who said that? I think it's adding a permanent allocation for nothing? (just curious, and surely wrong)

Reply:
clippy says:
```
warning: unnecessary closure used to substitute value for `Option::None`
   --> src/daemon/control.rs:513:23
    |
513 |               let sig = psbtin
    |  _______________________^
514 | |                 .partial_sigs
515 | |                 .get(&pubkey)
516 | |                 .ok_or_else(|| SigError::MissingSignature(pubkey))?;
    | |__________________________________________________________________^
    |
    = note: `#[warn(clippy::unnecessary_lazy_evaluations)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: Use `ok_or` instead
    |
513 |             let sig = psbtin
514 |                 .partial_sigs
515 |                 .get(&pubkey).ok_or(SigError::MissingSignature(pubkey))?;
```

I think the point is that, yes, you're adding a permanent allocation instead of re-evaluating a costant every time?

Reply:
Yes, makes sense